prereise/gather/griddata/hifld/data_process/topology.py

```diff
@@ -116,9 +194,20 @@ def connect_islands_with_minimum_cost(
             nodes2 = subs.loc[cc2].query("STATE in @cc2_adj")[["LATITUDE", "LONGITUDE"]]
             # Run an exhaustive search on the filtered substations of both connected
             # components to find a line with minimum cost the connects the two islands
-            min_dist, sub1, sub2 = min_dist_of_2_conn_comp(
-                nodes1, nodes2, cost_metric, memory_efficient=memory_efficient
-            )
+            if min_dist_method == "kdtree":
+                if kdtree_kwargs is None:
+                    kdtree_kwargs = {}
+                if kdtree_kwargs is not None and not isinstance(kdtree_kwargs, dict):
```

Comment:
At this point, we know that `kdtree_kwargs is not None`, because of the block immediately before, right?

Reply:
Done.